test/test_jit_fuser.py

```diff
@@ -121,8 +116,9 @@ def scaleshift(x, scale, shift):
 
     @unittest.skipIf(not RUN_CUDA, "fuser requires CUDA")
     @unittest.skipIf(not RUN_CUDA_HALF, "no half support")
-    @unittest.skipIf(graph_executor_mode() != ProfilingMode.LEGACY, "no half support with profiling on")
     def test_cuda_half(self):
+        if graph_executor_mode() == ProfilingMode.PROFILING:
```

Comment:
this might be actually a bit confusing since folks including me are used to looking for `skipIf` decorators. What's the reason for switching?

Reply:
The reason is that `skipIf` decorators are executed early and thus don't do what we expect them to do in this case. The checks `graph_executor_mode() == ...` might be executed not immediately before the test is run and thus result in incorrect behaviour.